tasks/main.yml

```
@@ -3,10 +3,16 @@
 - name: Set platform/version specific variables
   include_tasks: set_vars.yml
 
-- name: Ensure required packages are installed
-  package:
-    name: "{{ __podman_packages }}"
-    state: present
+- name: Gather the package facts
+  ansible.builtin.package_facts:
```

```suggestion
- name: Gather the package facts
  package_facts:

- name: Ensure required packages are installed
  package:
    name: "{{ __podman_packages }}"
    state: present
  become: true
  when: (__podman_packages | difference(ansible_facts.packages))
```
I know, I know - Ansible sez "must use FQCN everywhere, always!" - but the truth is that short names for builtins will be supported for the foreseeable future and then some, and the style of system roles is to use the short name, at least for builtins.
Also, default package manager is `auto` so can be omitted.

I think this will cover all cases
* Run as non-root, want to use sudo - will use sudo only for installing packages
* Run as non-root, do not want to use sudo - if podman packages are present, `package` should be skipped by the `when` condition
* Run as root - use `package` as normal

Does this work for you?